lib/cylc/network/suite_state.py

```diff
@@ -111,6 +123,22 @@ def update(self, tasks, tasks_rh, min_point, max_point, max_point_rh, paused,
 
         all_states.sort()
 
+        # Compute state_counts (total, and per cycle).
+        state_count_totals = {}
+        state_count_cycles = {}
+        for point_string, name_states in task_states.items():
+            count = {}
+            for name, state in name_states.items():
+                if state in count:
+                    count[state] += 1
+                else:
+                    count[state] = 1
+                if state in state_count_totals:
+                    state_count_totals[state] += 1
+                else:
+                    state_count_totals[state] = 1
+            state_count_cycles[point_string] = count
+
```

Comment:
(Note the above is a whole lot simpler than the code for generating state counts from ```cylc cat-state``` output).